pyfstat/utils/sft.py

```diff
@@ -140,31 +147,30 @@ def get_official_sft_filename(
     label: str or None
         optional 'Misc' entry in the SFT 'D' field
     window_type: str or None
-        included for SFT-spec v3 forwards compatibility
-        (see https://git.ligo.org/lscsoft/lalsuite/-/merge_requests/2027 );
-        not implemented yet
-    window_beta: float or None
-        included for SFT-spec v3 forwards compatibility
-        (see https://git.ligo.org/lscsoft/lalsuite/-/merge_requests/2027 );
-        not implemented yet
+        window function applied to SFTs
+    window_param: float or None
+        additional parameter for some window functions
 
     Returns
     -------
     filename: str
         The canonical SFT file name for the input parameters.
     """
-    if window_type or window_beta:
-        raise NotImplementedError(
-            "The parameters 'window_type' and 'window_beta'"
-            " are only included for SFT-spec v3 forwards compatibility"
-            " and not yet implemented."
-        )
-    return lalpulsar.OfficialSFTFilename(
-        IFO[0],
-        IFO[1],
-        numSFTs,
-        Tsft,
-        tstart,
-        duration,
-        label,
+    spec = lalpulsar.SFTFilenameSpec()
+    lalpulsar.FillSFTFilenameSpecStrings(
+        spec=spec,
+        path=None,
+        extn=None,
+        detector=IFO,
+        window_type=window_type,
+        privMisc=label,
+        pubObsKind=None,
+        pubChannel=None,
     )
+    spec.window_param = window_param or 0
+    spec.numSFTs = numSFTs
+    spec.SFTtimebase = Tsft
+    spec.gpsStart = tstart
+    # possible gotcha: duration may be different if nanoseconds of sft-epochs are non-zero
```

Comment:
Should we set a warning? LALPulsar codes don't work with nanoseconds in any case.

Reply:
I don't quite remember why I put this here; these are indeed UINT4 fields in https://lscsoft.docs.ligo.org/lalsuite/lalpulsar/struct_s_f_t_filename_spec.html#a2462122a0833c0d5d2084dc543d0c1bf not GPS structs, so it should just fail when given a float...?

Reply:
Indeed the SFT spec struct has UINT4 fields, but the actual SFTs have full GPS info theoreticlaly allowing for nanoseconds. Comments was echoed from https://lscsoft.docs.ligo.org/lalsuite/lalpulsar/_s_f_tfile_i_o_8c_source.html#l00675 . Extended to refer to it in ac1f5a9 .